feat: add teardown feed procedure to cleanup when not needing feed anymore

theophanemayaud opened the pull request:
When not needing a feed anymore (ex deleting the original table or disabling change feed on it), a few cleanups can be done.
This pr adds such a cleanup helper, to be called before or after deleting a change feed table, or just not needing the change feed on it.

theophanemayaud committed:
feat: add teardown feed procedure to cleanup when not needing feed anymore

theophanemayaud committed:
add documentation

starsheriff commented on migrations/2001.changefeed-v2.sql:
we should rename this to v3

theophanemayaud committed:
update test harness

theophanemayaud committed:
rename to v3 after change

theophanemayaud committed:
Merge remote-tracking branch 'origin/main' into feat-add-teardown-feed-to-cleanup

theophanemayaud committed:
Align tests and docs to v3 migration

Update Go fixtures and DbUp embedded resource to use the v3 migration file and import msdsn correctly so CI builds succeed.

theophanemayaud committed:
Normalize docker compose test file newline

Add trailing newline to docker-compose.test.yml for formatting consistency.
